src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeIT.java

```diff
@@ -355,4 +362,34 @@ void getEventOverviewsTest() throws NoCredentialsAvailableException, NoUserExcep
     assertEquals(8, numberOfPublicEventsInResults);
     assertEquals(1, numberOfPrivateEventsInResults);
   }
+
+  @Test
+  void cancelBookingAndPromoteWaitingListTest() throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+      AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+      AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+      MFARequiredButNotConfiguredException, SQLException {
+    String eventId = "dc8686cf-be3b-4c0d-8761-1e5504146867";
+    LoginResult studentLogin = loginAs(httpSession, ITConstants.TEST_STUDENT_EMAIL, ITConstants.TEST_STUDENT_PASSWORD);
+    HttpServletRequest cancelBookingRequest = createRequestWithCookies(new Cookie[] {studentLogin.cookie});
+    replay(cancelBookingRequest);
+    try (Response cancelBookingResponse = eventsFacade.cancelBooking(cancelBookingRequest, eventId)) {
+      assertEquals(Response.Status.NO_CONTENT.getStatusCode(), cancelBookingResponse.getStatus());
+    }
+
+    Map<Long, BookingStatus> bookingMap = new HashMap<>();
+    try (PreparedStatement pst = postgresSqlDb.getDatabaseConnection().prepareStatement(
```

Comment:
can you use a method in the codebase to pull this data back instead of performing a query? Just concerned that the test is tightly coupled to the db schema

Reply:
Should be possible. We do have a method or two that would be suitable but the class they're in doesn't appear to be made directly accessible by the integration tests currently.
I can have a look at adding that in and switching things over if that would be the preferred approach.

Reply:
I wouldn't change visibility of any code to facilitate this - having sql code in the test is the lesser of those two evils.